woocommerce/woocommerce
# Create schedule sale pricing block

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



Closes https://github.com/woocommerce/woocommerce/issues/37394



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Got to /wp-admin/tools.php?page=woocommerce-admin-test-helper
2. Under Features tab make sure to enable block-editor-feature-enabled
3. Then visit /wp-admin/admin.php?page=wc-admin&path=/add-product
4. Under the Pricing tab in the Pricing section the schedule sale fields should be shown
5. Schedule sale toggle should be present within pricing block in pricing tab
6. Schedule sale toggle should be disabled when no sale price exists
7. When enabled the `from` and `to` date time fields should be present
8. When disabled the `from` and `to` field values should be set back to blank
9. When initially enabled the `from` field should be auto populated to the present day

https://user-images.githubusercontent.com/13334210/229915406-ab227f2b-34ae-4a38-b640-525eb9306fff.mov
